src/rust/engine/options/src/args_tests.rs

```diff
@@ -180,6 +180,16 @@ fn test_scalar_fromfile() {
     do_test("-42", -42, Args::get_int, false);
     do_test("3.14", 3.14, Args::get_float, false);
     do_test("EXPANDED", "EXPANDED".to_owned(), Args::get_string, false);
+
+    let (_tmpdir, fromfile_path) = write_fromfile("fromfile.txt", "BAD INT");
+    let args = Args {
+        args: vec![format!("--foo=@{}", fromfile_path.display())],
+    };
+    assert_eq!(
+        args.get_int(&option_id!("foo")).unwrap_err(),
+        "Problem parsing --foo int value:\n1:BAD INT\n  ^\n\
+               Expected \"+\", \"-\" or ['0'..='9'] at line 1 column 1"
```

Comment:
Is it possible to reference the file path here too?

It probably doesn't matter so much for arguments, where the file name is likely to be directly visible in the invocation, but is potentially more useful for:

-  values from an environment variable (to save a user from having to work out the env var value, which can be annoying for non-interactive executions like CI)
- values that have `%(...)` interpolation placeholders in them (if that's a thing that works for file paths).

Reply:
Every error references the path (and option ID). It gets added at a higher level, in the OptionParser (by mk_parse_err).